blocks/filesystem.py

```diff
@@ -147,32 +146,86 @@ def ls(self, path):
             output = glob.glob(path)
         return sorted(p.rstrip('/') for p in output)
 
-    def copy(self, sources, dest):
-        """ Copy the files in sources (recursively) to dest
+    def rm(self, paths, recursive=False):
+        """ Remove the files at paths
+
+        Parameters
+        ----------
+        paths : list of str
+            The paths to remove
+        recursive : bool, default False
+            If true, recursively remove any directories
+        """
+        if isinstance(paths, string_types):
+            paths = [paths]
+
+        if any(not self.local(p) for p in paths):
+            # at least one location is on GCS
+            cmd = ['gsutil', '-m', 'rm']
+
+        else:
+            cmd = ['rm']
+
+        if recursive:
+            cmd.append('-r')
+
+        CHUNK_SIZE = 1000
+        paths_chunks = [paths[x:x+CHUNK_SIZE] for x in range(0, len(paths), CHUNK_SIZE)]
+        for paths in paths_chunks:
+            subprocess.check_call(cmd + paths)
+
+    def cp(self, sources, dest, recursive=False):
+        """ Copy the files in sources to dest
 
         Parameters
         ----------
         sources : list of str
-            The list of paths to copy, which can be directories
+            The list of paths to copy
         dest : str
             The destination for the copy of source(s)
+        recursive : bool
+            If true, recursively copy any directories
         """
+        if isinstance(sources, string_types):
+            sources = [sources]
+
         summary = ', '.join(sources)
         logging.info('Copying {} to {}...'.format(summary, dest))
 
-        # TODO perhaps use a python API rather than subprocess
         if any(self.GCS in x for x in sources + [dest]):
             # at least one location is on GCS
             cmd = self.gcscp
         else:
-            cmd = ['cp', '-r']
-        # Break this into pieces so we don't hit OS limit for arguments
-        # TODO: use getconf ARG_MAX and actually test number of bytes in sources
+            cmd = ['cp']
+
+        if recursive:
+            cmd.append('-r')
+
         CHUNK_SIZE = 1000
         sources_chunks = [sources[x:x+CHUNK_SIZE] for x in range(0, len(sources), CHUNK_SIZE)]
         for sources in sources_chunks:
             subprocess.check_call(cmd + sources + [dest])
 
+    @contextmanager
+    def open(self, path, mode='rb'):
+        """ Access path as a file-like object
+
+        Parameters
+        ----------
+        path: str
+            The path of the file to access
+
```

Comment:
Need a `mode` parameter in the docstring here?

Reply:
✅